docs/en/sql-reference/functions/ext-dict-functions.md

```diff
@@ -7,226 +7,188 @@ sidebar_label: Dictionaries
+If ClickHouse parses the attribute successfully in the [attribute’s data type](../../sql-reference/dictionaries/index.md#dictionary-key-and-fields#ext_dict_structure-attributes), this function returns the value of the dictionary attribute that corresponds to `id_expr`.
 
-- If there is no the key, corresponding to `id_expr`, in the dictionary, then:
+If there is no the key, corresponding to `id_expr`, in the dictionary, then this function returns the content of the `<null_value>` element specified for the attribute in the dictionary configuration.
 
-        - `dictGet` returns the content of the `<null_value>` element specified for the attribute in the dictionary configuration.
-        - `dictGetOrDefault` returns the value passed as the `default_value_expr` parameter.
-        - `dictGetOrNull` returns `NULL` in case key was not found in dictionary.
+**Implementation details**
 
 ClickHouse throws an exception if it cannot parse the value of the attribute or the value does not match the attribute data type.
 
-**Example for simple key dictionary**
+**Example**
 
-Create a text file `ext-dict-test.csv` containing the following:
+Query:
 
-``` text
-1,1
-2,2
-```
-
-The first column is `id`, the second column is `c1`.
-
-Configure the dictionary:
-
-``` xml
-<clickhouse>
-    <dictionary>
-        <name>ext-dict-test</name>
-        <source>
-            <file>
-                <path>/path-to/ext-dict-test.csv</path>
-                <format>CSV</format>
-            </file>
-        </source>
-        <layout>
-            <flat />
-        </layout>
-        <structure>
-            <id>
-                <name>id</name>
-            </id>
-            <attribute>
-                <name>c1</name>
-                <type>UInt32</type>
-                <null_value></null_value>
-            </attribute>
-        </structure>
-        <lifetime>0</lifetime>
-    </dictionary>
-</clickhouse>
-```
-
-Perform the query:
+```sql
+CREATE TABLE source_table
+(
+    id UInt64,
+    value String
+) ENGINE = MergeTree
+ORDER BY id;
 
-``` sql
-SELECT
-    dictGetOrDefault('ext-dict-test', 'c1', number + 1, toUInt32(number * 10)) AS val,
-    toTypeName(val) AS type
-FROM system.numbers
-LIMIT 3;
+INSERT INTO source_table VALUES (1, 'First'), (2, 'Second');
+
+CREATE DICTIONARY id_value_dictionary
+(
+    id UInt64,
+    value String
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(TABLE 'source_table'))
+LIFETIME(MIN  0 MAX  1000)
+LAYOUT(FLAT());
+
+SELECT dictGet('id_value_dictionary', 'value', toUInt64(1)) AS value;
 ```
 
-``` text
-┌─val─┬─type───┐
-│   1 │ UInt32 │
-│   2 │ UInt32 │
-│  20 │ UInt32 │
-└─────┴────────┘
+**Result**:
+
+```response
+'First'
 ```
 
-**Example for complex key dictionary**
+## dictGetOrDefault
 
-Create a text file `ext-dict-mult.csv` containing the following:
+Retrieves values from a dictionary. If that value cannot be found, returns the default value set by the user.
```

Comment:
I think it's not good to say `If that value cannot be found`, because we search values by keys, and if we cannot find the key, then the default value will be set

Reply:
I don't understand. Please leave a suggestion.

Reply:
```suggestion
Retrieves values from a dictionary. Returns the default value set by the user for keys that are not found in the dictionary.
```